File: Android/app/src/main/java/com/edu/primary/database/dao/AnswerDao.java
package com.edu.primary.database.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import com.edu.primary.database.entity.AnswerEntity;

import java.util.List;

@Dao
public interface AnswerDao {
    @Insert
    long insertAnswer(AnswerEntity answer);

    @Query("SELECT * FROM answers WHERE userId = :userId")
    List<AnswerEntity> getAnswersByUser(long userId);

    @Query("SELECT * FROM answers WHERE userId = :userId AND questionId = :questionId")
    AnswerEntity getAnswer(long userId, long questionId);

    @Query("SELECT COUNT(*) FROM answers WHERE userId = :userId AND isCorrect = 1")
    int getCorrectCount(long userId);

    @Query("SELECT COUNT(*) FROM answers WHERE userId = :userId")
    int getTotalCount(long userId);
}

File: Android-1/app/src/main/java/com/edu/primary/data/local/dao/AnswerDao.kt
package com.edu.primary.data.local.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.edu.primary.data.local.entity.AnswerEntity

@Dao
interface AnswerDao {
    @Insert
    suspend fun insert(answer: AnswerEntity): Long

    @Query("SELECT * FROM answers WHERE userId = :userId")
    suspend fun getByUser(userId: Long): List<AnswerEntity>

    @Query("SELECT * FROM answers WHERE userId = :userId AND questionId = :questionId")
    suspend fun get(userId: Long, questionId: Long): AnswerEntity?

    @Query("SELECT COUNT(*) FROM answers WHERE userId = :userId AND isCorrect = 1")
    suspend fun getCorrectCount(userId: Long): Int

    @Query("SELECT COUNT(*) FROM answers WHERE userId = :userId")
    suspend fun getTotalCount(userId: Long): Int
}
